src/cloudai/schema/test_template/nccl_test/kubernetes_json_gen_strategy.py

```diff
@@ -0,0 +1,262 @@
+            },
+        }
+
+    def _generate_env_list(self, env_vars: Dict[str, str]) -> List[Dict[str, str]]:
+        """
+        Generate the environment variables list for the Kubernetes container.
+
+        Args:
+            env_vars (Dict[str, str]): A dictionary of environment variables to include.
+
+        Returns:
+            List[Dict[str, str]]: A list of dictionaries representing the environment variables.
+        """
+        env_list = [{"name": "OMPI_ALLOW_RUN_AS_ROOT", "value": "1"}]
+        # Include additional environment variables from env_vars
+        for key, value in env_vars.items():
+            env_list.append({"name": key, "value": value})
+        return env_list
+
+    def _generate_launcher_command(
+        self,
+        final_num_nodes: int,
+        nodes: List[str],
+        env_vars: Dict[str, str],
+        cmd_args: Dict[str, str],
+        extra_cmd_args: str,
+    ) -> List[str]:
+        """
+        Generate the launcher command for the Kubernetes container.
+
+        Args:
+            final_num_nodes (int): The final number of nodes determined.
+            nodes (List[str]): A list of specific nodes to run the job on.
+            env_vars (Dict[str, str]): A dictionary of environment variables for the job.
+            cmd_args (Dict[str, str]): A dictionary of command-line arguments for the NCCL test.
+            extra_cmd_args (str): Additional command-line arguments for the NCCL test.
+
+        Returns:
+            List[str]: A list representing the launcher command to be executed.
+        """
+        subtest_name = cmd_args.get("subtest_name")
+        if subtest_name is None:
+            raise ValueError(
+                "The NCCL test's 'subtest_name' is not provided. Please ensure 'subtest_name' "
+                "is included in the command arguments. Valid subtest names include: "
+                "all_reduce_perf, all_gather_perf, alltoall_perf, broadcast_perf, "
+                "gather_perf, hypercube_perf, reduce_perf, reduce_scatter_perf, "
+                "scatter_perf, and sendrecv_perf."
+            )
+
+        nccl_test_args = [
+            "nthreads",
+            "ngpus",
+            "minbytes",
+            "maxbytes",
+            "stepbytes",
+            "op",
+            "datatype",
+            "root",
+            "iters",
+            "warmup_iters",
+            "agg_iters",
+            "average",
+            "parallel_init",
+            "check",
+            "blocking",
+            "cudagraph",
+        ]
+
+        command_parts = [f"/opt/nccl_tests/build/{subtest_name}"]
+        for arg in nccl_test_args:
+            if arg in cmd_args:
+                command_parts.append(f"--{arg} {cmd_args[arg]}")
+
+        if extra_cmd_args:
+            command_parts.append(extra_cmd_args)
+
+        return [
+            "/bin/bash",
+            "-c",
+            (
+                f"mpirun -v --allow-run-as-root -np {final_num_nodes} "
```

Comment:
`--allow-run-as-root` Does CloudAI or user need to be added in some specific group to have this privilege? In babyx we kind of own the cluster. But is this assumption always true?

Reply:
`allow-run-as-root` is required because you are running as root in the containers. It is not related to the host's permissions or privileges.